### DISCUSSION TITLE
운영체제 - 1. 소개

### DISCUSSION THREAD

**Discussion by eatingbug** [Team-Psyduck/tech-sharing#6] [Operating System]
- 강의소개
- Introduction to Operating Systems
  - 운영체제란 무엇인가, 운영체제의 목적, 운영체제의 분류, 운영체제의 예, 운영체제의 구조

**Comment by kkg5** [comment]
![image](https://user-images.githubusercontent.com/97646802/216979368-d9eda67b-d072-42a5-9dd6-f1b96cbe37fa.png)

## 운영체제의 목적
- 자원을 `효율적`으로 `형평성` 있게 관리
- 사용자 및 운영체제 자신을 `보호`
- 사용자에게 `편리성` 제공

## 운영체제의 분류
### 1. 동시 작업 가능 여부
- 단일 작업 (Single Tasking)
- 다중 작업 (Multi Tasking) - `현대의 운영체제`
### 2. 사용자의 수
- 단일 사용자 (Single User)
- 다중 사용자 (Multi User)
### 3. 처리 방식
- 일괄 처리 (Batch Processing)
- 시분할 (Time Sharing) - `우리의 컴퓨터`
- 실시간 (Realtime OS)

## 운영체제의 예
### 1. UNIX
- 코드의 대부분을 `C언어`로 작성
- 높은 `이식성`
- 최소한의 커널 구조
- 소스 코드 공개
- 확장 및 프로그램 개발에 용이
- 다양한 확장 버전
### 2. DOS (Disk Operating System)
- MS사
- 단일 사용자용 운영체제
### 3. MS Windows
- MS사
- 다중 작업용 GUI 기반 운영체제
### 4. 기타
- Handheld Device를 위한 OS
- iOS, AOS

## 운영체제의 구조
![image](https://user-images.githubusercontent.com/97646802/216995840-c60876d2-3b33-4790-a471-f5f46432007b.png)

**Comment by eatingbug** [comment]
## 강의소개
- 운영체제란 컴퓨터 하드웨어 바로 위에 설치되는 소프트웨어 계층으로 모든 컴퓨터 시스템의 필수적인 부분이다.
- 운영체제의 목적은 컴퓨터 시스템의 자원을 효율적으로 관리하는 것이다.
- 본 강의 목표는 운영체제의 개념과 역할, 운영체제를 구성하는 요소와 그 알고리즘의 핵심을 학습하는 것이다.

## 운영체제 소개

### 운영체제 분류
- 동시 작업 가능 여부
  - 단일 작업
  - 다중 작업
- 사용자 수
  - 단일 사용
  - 다중 사용
- 처리 방식
  - 일괄 처리
  - 시분할 : 우리의 운영체제
  - 실시간 : 특수목적

### 운영체제 예
- UNIX
  - 대형 컴퓨터 용도로 만들어져서 다중 작업, 다중 사용 가능
  - 대부분이 C언어로 만들어짐
  - 오픈소스
  - 높은 이식성
  - Linux
- DOS
- MS Windows

### 운영체제의 구조
- CPU 스케줄링
- 메모리 관리
- 파일 관리
- 입출력 관리 (인터럽트)
- 프로세스

**Comment by HeoJunYeol** [comment]
# Introduction to Operating Systems

### 운영 체제(Operating System, OS)란?
- 컴퓨터 하드웨어 바로 위에 설치되어 사용자 및 다른 모든 소프트웨어와 하드웨어를 연결하는 소프트웨어 계층

### 운영 체제 목적
 - 자원을 효율적으로 관리
 - 사용자 편의성 제공
 - 사용자 및 운영체제 자신의 보호

### 운영 체제 분류
 - 동시 작업 가능 여부
 - 사용자의 수
 - 처리 방식
   - 일괄 처리
   - 시분할 처리
   - 실시간 처리

### 운영 체제 종류
#### 유닉스(UNIX)
- 코드의 대부분 C언어로 작성
- 높은 이식성
- 최소한의 커널 구조
- 복잡한 시스템에 맞게 확장 용이
- 프로그램 개발에 용이
- 소스 코드 공개
- 다양한 버전

#### DOS(Disk Operating System)
- MS사에서 1981년 IBM-PC를 위해 개발
- 단일 사용자용 운영체제, 메모리 관리 능력의 한계 (640KB)

#### MS Windows
- MS사의 다중 작업용 GUI 기반 운영 체제
- Plug and Play, 네트워크 환경 강화
- DOS용 응용 프로그램과의 호환성 제공
- 불안정성
- 풍부한 지원 소프트웨어

#### Handheld Devicd를 위한 OS
- PalmOS, Pocket PC, Tiny OS

### 운영 체제 구조
- CPU 스케줄링
- 메모리 관리
- 파일 관리
- 입출력 관리
- 프로세스 관리
